src/utils/gethRead.js

```diff
@@ -10,19 +10,27 @@ async function send({ method, params }, provider) {
 }
 
 function fetchBlock(blockNumber, includeFullTransactions = true) {
-  const method = {
-    method: 'eth_getBlockByNumber',
-    params: [ethers.utils.hexlify(blockNumber), includeFullTransactions], // [hex block number, include full transactions boolean]
-  }
-  return send(method).then(parseBlock)
+  try {
+    const method = {
+      method: 'eth_getBlockByNumber',
+      params: [ethers.utils.hexlify(blockNumber), includeFullTransactions], // [hex block number, include full transactions boolean]
+    }
+    return send(method).then(parseBlock)
+  } catch (e) {
+    catchError(e)
```

Comment:
What is this `catchError` referencing?

Reply:
Ah never mind I see it defined below. I’m now sure what this code adds, since it catches the error and then throws it.

Reply:
If the intention is to bubble up an error inside a promise this isn’t going to cut it I don’t think

Reply:
Right now, I'm seeing logs that suggest that these promises, when they don't resolve cleanly, lead to an UnhandledPromiseRejectionWarning. I want to make that explicitly thrown as an error so it crashes the pod I'm running instead of leaving the pod in a dead state where Kubernetes thinks it's still running.